client/my-sites/activity/activity-log-tasklist/to-update.jsx

```diff
@@ -1,67 +1,59 @@
 
-const emptyList = [];
+const EMPTY_LIST = [];
 
 const unionBySlug = ( a = [], b = [] ) => [
 	...a,
 	...b.filter( ( be ) => ! a.some( ( ae ) => ae.slug === be.slug ) ),
 ];
 
-export default ( WrappedComponent ) => {
-	class ToUpdate extends Component {
-		static propTypes = {
-			siteId: PropTypes.number,
+export default function ( WrappedComponent ) {
+	function WithItemsToUpdate( { siteId, ...props } ) {
+		const pluginsWithUpdates = useSelector( ( state ) => {
+			if ( isJetpackSiteSecondaryNetworkSite( state, siteId ) ) {
+				return EMPTY_LIST;
+			}
+			return getPluginsWithUpdates( state, [ siteId ] ) ?? EMPTY_LIST;
+		} );
 
-			// Connected
-			plugins: PropTypes.arrayOf( PropTypes.object ),
-			themes: PropTypes.arrayOf( PropTypes.object ),
-			core: PropTypes.arrayOf( PropTypes.object ),
-		};
+		// merge the `pluginsWithUpdates` object into the previous value, so that plugins that
+		// have been updated are still part of the list and don't disappear.
+		const plugins = useRef( { siteId: undefined, mergedPlugins: undefined } );
+		const mergedPlugins =
+			plugins.current.siteId === siteId
+				? unionBySlug( pluginsWithUpdates, plugins.current.mergedPlugins )
+				: pluginsWithUpdates;
+		plugins.current = { siteId, mergedPlugins };
 
-		state = {
-			// Plugins already updated + those with pending updates
-			plugins: emptyList,
-			siteId: this.props.siteId,
-		};
+		const siteAlertsQuery = useQuery(
+			[ 'site-alerts', siteId ],
+			() => wpcom.req.get( { path: `/sites/${ siteId }/alerts`, apiNamespace: 'wpcom/v2' } ),
+			{ refetchInterval: 5 * 60 * 1000 }
+		);
 
-		static getDerivedStateFromProps( nextProps, prevState ) {
-			return {
-				plugins:
-					nextProps.siteId === prevState.siteId
-						? unionBySlug( nextProps.plugins ?? [], prevState.plugins ?? [] )
-						: emptyList,
-				siteId: nextProps.siteId,
-			};
-		}
+		const themes = siteAlertsQuery.data?.updates?.themes ?? EMPTY_LIST;
+		const core = siteAlertsQuery.data?.updates?.core ?? EMPTY_LIST;
 
-		render() {
-			return (
-				<WrappedComponent
-					{ ...this.props }
-					siteId={ this.props.siteId }
-					plugins={ this.state.plugins }
-					themes={ this.props.themes }
-					core={ this.props.core }
-				/>
-			);
-		}
+		return (
+			<WrappedComponent
+				{ ...props }
+				siteId={ siteId }
+				plugins={ mergedPlugins }
+				themes={ themes }
+				core={ core }
+			/>
+		);
 	}
-	return connect( ( state, { siteId } ) => {
-		const alertsData = requestSiteAlerts( siteId );
-		let pluginsWithUpdates = emptyList;
-		if ( ! isJetpackSiteSecondaryNetworkSite( state, siteId ) ) {
-			pluginsWithUpdates = getPluginsWithUpdates( state, [ siteId ] );
-		}
-		return {
-			plugins: pluginsWithUpdates,
-			themes: get( alertsData, 'data.updates.themes', emptyList ),
-			core: get( alertsData, 'data.updates.core', emptyList ),
-		};
-	} )( ToUpdate );
-};
+
+	WithItemsToUpdate.propTypes = {
+		siteId: PropTypes.number,
```

Comment:
Isn't the `siteId` prop required?

Reply:
Yep, added an `.isRequired` suffix there :+1: